echo-pipelinetriggers/src/main/java/com/netflix/spinnaker/echo/pipelinetriggers/artifacts/ArtifactMatcher.java

```diff
@@ -39,7 +39,16 @@ public static Boolean anyArtifactsMatchExpected(List<Artifact> messageArtifacts,
       return true;
     }
 
-    List<ExpectedArtifact> pipelineExpectedArtifacts = pipeline.getExpectedArtifacts();
+    List<ExpectedArtifact> pipelineExpectedArtifacts = pipeline.getExpectedArtifacts()
+        .stream()
+        .filter(e -> !e.isUsePriorArtifact())
+        .collect(Collectors.toList());
+
+    if ( pipelineExpectedArtifacts == null || pipelineExpectedArtifacts.isEmpty()) {
+      log.info("All pipeline expected artifacts are marked as UsePrior hence no need for matching.");
+      return true;
+    }
+
     List<ExpectedArtifact> expectedArtifacts = pipelineExpectedArtifacts == null ? new ArrayList<>() : pipelineExpectedArtifacts
```

Comment:
I think the real fix is to return 'true' when this is empty. The reason being, the "anyMatch" on the below line here returns "false" when the list of expected artifacts to check is empty: https://github.com/spinnaker/echo/pull/269/files#diff-414234a528208013567f43f82f05dc76R64

Reply:
Unless someone ends up messing with the expected artifacts ids manually, `expectedArtifacts` should not ever be empty.
`expectedArtifacts` should always be a subset of `pipelineExpectedArtifacts` and it's "emptiness" is already checked in here: https://github.com/PerGon/echo/blob/bda18a6d1d59e2e39568040fdd39da549e391df3/echo-pipelinetriggers/src/main/java/com/netflix/spinnaker/echo/pipelinetriggers/artifacts/ArtifactMatcher.java#L38
(it might happen that due to some other bug, the `expectedArtifacts` will be empty, so that check should be added. But that not the case I'm trying to build :) will try to explain better in a comment)

Reply:
Well if that's the case, and trigger 2 does not expect any artifacts -- isn't this already working as intended?

Reply:
After realizing I was messing up, yes, all works fine now! Sorry :D